packages/smart-wallet/src/proposals/upgrade-walletFactory-proposal.js

```diff
@@ -0,0 +1,252 @@
+    walletFactoryRef.bundleID,
+    newPrivateArgs,
+  );
+  console.log('upgradeWalletFactory: done');
+};
+harden(upgradeWalletFactory);
+
+/**
+ * @param { BootstrapPowers } powers
+ */
+export const publishAgoricBrandsDisplayInfo = async ({
+  consume: { agoricNames, board, chainStorage },
+}) => {
+  // @ts-expect-error chainStorage is only falsy in testing
+  const boardAux = E(chainStorage).makeChildNode(BOARD_AUX);
+  const publishBrandInfo = async brand => {
+    const [id, displayInfo, allegedName] = await Promise.all([
+      E(board).getId(brand),
+      E(brand).getDisplayInfo(),
+      E(brand).getAllegedName(),
+    ]);
+    const node = E(boardAux).makeChildNode(id);
+    const aux = marshalData.toCapData(harden({ allegedName, displayInfo }));
+    await E(node).setValue(JSON.stringify(aux));
+  };
+
+  /** @type {ERef<NameHub>} */
+  const brandHub = E(agoricNames).lookup('brand');
+  const brands = await E(brandHub).values();
+  // tolerate failure; in particular, for the timer brand
+  await Promise.allSettled(brands.map(publishBrandInfo));
+};
+harden(publishAgoricBrandsDisplayInfo);
+
+/** @type { import("@agoric/vats/src/core/lib-boot").BootstrapManifest } */
+const manifest = {
+  [upgradeWalletFactory.name]: {
+    // include rationale for closely-held, high authority capabilities
+    consume: {
+      contractKits: `to upgrade walletFactory using its adminFacet`,
+      governedContractKits:
+        'to get walletReviver from provisionPool.creatorFacet',
+      chainStorage: 'to allow walletFactory to (continue) write to vstorage',
+      walletBridgeManager: 'to handle bridged cosmos SpendAction messages',
+    },
+    // widely-shared, low authority instance handles need no rationale
+    instance: {
+      consume: { walletFactory: true, provisionPool: true },
+    },
+  },
+  [publishAgoricBrandsDisplayInfo.name]: {
+    consume: { agoricNames: true, board: true, chainStorage: true },
+  },
+};
+harden(manifest);
+
+export const getManifestForUpgrade = (_powers, { walletFactoryRef }) => {
+  return harden({
+    manifest,
+    options: { walletFactoryRef },
+  });
+};
+
+// avoid importing all of ERTP
+const makeAmount = (brand, value) => harden({ brand, value });
+
+const IST_UNIT = 1_000_000n;
+const CENT = IST_UNIT / 100n;
+
+/**
+ * Make a storage node for auxilliary data for a value on the board.
+ *
+ * @param {ERef<StorageNode>} chainStorage
+ * @param {string} boardId
+ */
+const makeBoardAuxNode = async (chainStorage, boardId) => {
+  const boardAux = E(chainStorage).makeChildNode(BOARD_AUX);
+  return E(boardAux).makeChildNode(boardId);
+};
+
+const publishBrandInfo = async (chainStorage, board, brand) => {
+  const [id, displayInfo] = await Promise.all([
+    E(board).getId(brand),
+    E(brand).getDisplayInfo(),
+  ]);
+  const node = makeBoardAuxNode(chainStorage, id);
+  const aux = marshalData.toCapData(harden({ displayInfo }));
+  await E(node).setValue(JSON.stringify(aux));
```

Comment:
This is creating a parallel hierarchy for board-published brands, right? And with a path that is keyed by boardID but lacks an obvious connection to "agoricNames"? Have the right people considered this design and possible alternatives?

Reply:
Yes, that's what it's doing. Whether it's sufficiently considered is a good question.

Design considerations included:

  - ideally, it would be a platform thing, but #2069 is an open research problem
  - should the mechanism be for things in agoricNames only or for any brand (or any other remotable, for that matter)?
    - we chose: any remotable on the board
  - should all the aux data for well-known brands be at one key like `agoricNames.vbankAsset` or should each brand have a separate key?
    - we chose a separate key

Turadg and I discussed those alternatives and agreed to this one. Sam found it inconvenient to make more than one query but eventually agreed it was better than doing O(n) work for each write.

Figuring out who are the right people is an ongoing challenge, for me. I think bringing in your perspective is useful.

Reply:
cc @erights

Reply:
In [today's office hours](https://github.com/Agoric/agoric-sdk/discussions/8240), I walked thru this with @erights . We determined that this doesn't suffer the worst part of the auxiliary data problem, which is that aux data passes thru a non-authoritative vat on the way from the authoritative object/vat to the consumer. In this `boardAux` design, the party that can write to `published.boardAux.boardNNN` is authoritative. Their authority derives from the way storage nodes are delegated in bootstrap and core-eval scripts.

He agreed this aspect of this PR is OK.